src/library_html5.js

```diff
@@ -400,8 +400,8 @@ var LibraryJSEvents = {
 
         var nodeName = JSEvents.getNodeNameForTarget(e.target);
         var id = e.target.id ? e.target.id : '';
-        writeStringToMemory(nodeName, JSEvents.focusEvent + {{{ C_STRUCTS.EmscriptenFocusEvent.nodeName }}} );
-        writeStringToMemory(id, JSEvents.focusEvent + {{{ C_STRUCTS.EmscriptenFocusEvent.id }}} );
+        stringToUTF8(nodeName, JSEvents.focusEvent + {{{ C_STRUCTS.EmscriptenFocusEvent.nodeName }}}, {{{ cDefine('EM_HTML5_LONG_STRING_LEN_BYTES') }}});
+        stringToUTF8(id, JSEvents.focusEvent + {{{ C_STRUCTS.EmscriptenFocusEvent.id }}}, {{{ cDefine('EM_HTML5_LONG_STRING_LEN_BYTES') }}});
```

Comment:
The uses of `writeStringToMemory` in `src/library_html5.js` all had an assumption that all string fields in the structs would have enough space that browsers can't possibly produce too long strings to overflow. While that assumption definitely holds for short strings like `key` and `char` fields, it is possible for DOM elements to have arbitrary long ids, here we now truncate after 128 bytes.